<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME)"/>
  <arg name="NATO_NAME" default="$(optenv NATO_NAME)"/>
  <arg name="UAV_TYPE" default="$(optenv UAV_TYPE)"/>
  <arg name="SENSORS" default="$(optenv SENSORS)"/>
  <arg name="RUN_TYPE" default="$(optenv RUN_TYPE)"/>
  <arg name="PIXGARM" default="$(optenv PIXGARM false)"/>
  <arg name="COLORSCHEME" default="$(optenv PROFILES_BOTH COLORSCHEME_LIGHT)"/>
  <arg name="RAINBOW" default="false"/>

  <arg name="debug" default="false"/>

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg name="config_file" default=""/>

  <arg if="$(eval arg('RUN_TYPE') == 'uav')" name="UAV_MASS" default="$(env UAV_MASS)" />

  <group ns="$(arg UAV_NAME)">

    <node name="mrs_uav_status_acquisition" pkg="mrs_uav_status" type="data_acquisition" output="screen" respawn="true" launch-prefix="$(arg launch_prefix)">
        
      <param name="uav_name" value="$(arg UAV_NAME)" />
      <param name="nato_name" value="$(arg NATO_NAME)" />
      <param name="uav_type" value="$(arg UAV_TYPE)" />
      <param if="$(eval arg('RUN_TYPE') == 'uav')" name="uav_mass" value="$(arg UAV_MASS)" />
      <param name="run_type" value="$(arg RUN_TYPE)" />
      <param name="sensors" value="$(arg SENSORS)" />
      <param name="pixgarm" type="bool" value="$(arg PIXGARM)" />

      <rosparam file="$(find mrs_uav_status)/config/default.yaml" command="load" />
      <rosparam if="$(eval arg('RUN_TYPE') == 'simulation')" file="$(find mrs_uav_managers)/config/$(arg RUN_TYPE)/$(arg UAV_TYPE)/mass.yaml" />
      <rosparam if="$(eval not arg('config_file') == '')" file="$(arg config_file)" />

        <!-- Subscribers -->
      <remap from="~uav_state_in" to="estimation_manager/uav_state" />
      <remap from="~cmd_position_in" to="control_manager/position_cmd" />
      <remap from="~odom_diag_in" to="estimation_manager/diagnostics_legacy" />
      <remap from="~mpc_diag_in" to="control_manager/mpc_tracker/diagnostics" />
      <remap from="~mavros_state_in" to="mavros/state" />
      <remap from="~cmd_attitude_in" to="control_manager/attitude_cmd" />
      <remap from="~battery_in" to="mavros/battery" />
      <remap from="~control_manager_in" to="control_manager/diagnostics" />
      <remap from="~gain_manager_in" to="gain_manager/diagnostics" />
      <remap from="~constraint_manager_in" to="constraint_manager/diagnostics" />
      <remap from="~tf_static_in" to="/tf_static" />
      <remap from="~mavros_global_in" to="mavros/global_position/global" />
      <remap from="~mavros_local_in" to="mavros/local_position/pose" />
      <remap from="~automatic_start_in" to="automatic_start/can_takeoff" />

      <remap from="~string_in" to="mrs_uav_status/display_string" />
      <remap from="~set_service_in" to="mrs_uav_status/set_trigger_service" />

        <!-- Publishers -->
      <remap from="~uav_status_out" to="mrs_uav_status/uav_status" />
      <remap from="~uav_status_short_out" to="mrs_uav_status/uav_status_short" />
      <remap from="~profiler" to="profiler" />


    </node>

  </group>

</launch>
